src/types/dict.rs

```diff
@@ -470,15 +478,15 @@ impl DictIterImpl {
                 let mut key: *mut ffi::PyObject = std::ptr::null_mut();
                 let mut value: *mut ffi::PyObject = std::ptr::null_mut();
 
-                if unsafe { ffi::PyDict_Next(dict.as_ptr(), ppos, &mut key, &mut value) } != 0 {
+                if ffi::PyDict_Next(dict.as_ptr(), ppos, &mut key, &mut value) != 0 {
```

Comment:
IMO we should keep these `unsafe` blocks. `unsafe_op_in_unsafe_fn` will be warn be default in the 2024 edition and it makes it easier validate and document the safety invariants.

Reply:
Thanks! For some reason I thought there was a clippy lint to remove these.